salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +50,94 @@ def node_info(node, ca_minion):
     return result
 
 
+def cluster_version_info(namespace_information):
+    cluster_version = None
+
+    if namespace_information:
+        ns_info_dict = namespace_information.to_dict()
```

Comment:
Why not use the object itself? I guess the whole method could benefit from a simpler syntax. Also, the `cluster-version` annotation is specific to the `kube-system` namespace, there's no real value in passing any namespace object.

```python
def get_cluster_version():
    namespace = coreV1.read_namespace(name="kube-system")
    annotation_key = 'metalk8s.scality.com/cluster-version'

    if annotation_key not in namespace.metadata.annotations:
        # return pillar `_errors`

    return namespace.metadata.annotations[annotation_key]
```